lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -2151,6 +2153,20 @@ static void transMetadataDecorations(Metadata *MD, SPIRVEntry *Target) {
           Target, Name->getString().str(), TypeKind));
       break;
     }
+    case spv::internal::DecorationHostAccessINTEL: {
```

Comment:
According to spec: "Only valid on global (module scope) OpVariable". It seems like there is no appropriate check for this.

Reply:
I thought that `transMetadataDecorations()` is only called for GlobalVariable: https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/a96f8da5942d175645b26f822929ea995fc31f26/lib/SPIRV/SPIRVWriter.cpp#L2268-L2271
But I found that it also can be called for kernel parameter decorations. 
Thanks for noticing, I'll add the appropriate check.